File: 1 - INICIANTE/1164.cpp
#include <iostream>
#include <iomanip>

using namespace std;

int somadosdivisores(int num);
int numeroperfeito(int num);

int main() {
  int i,c,num;
  cin >> c;

  for(i=0;i<c;i++) {
   cin >> num;
   if(numeroperfeito(num)==1) {
     cout << num << " eh perfeito" << endl;
   }
   else {
     cout << num << " nao eh perfeito" << endl;
   }
  }

return 0;


}

int somadosdivisores(int num) {
  int i,soma=0;
  for(i=1;i<=(num/2);i++) 
    if(num%i==0) 
      soma = soma + i;
      return soma;
}


int numeroperfeito(int num) {
  if (somadosdivisores(num)==num) {
    return 1;
  }
  else {
    return 0;
  } 

}

File: 1 - INICIANTE/python/1164.py
n = int(input())
while n != 0:
    x = int(input())
    soma = 0
    for c in range(1, x):
        if x % c == 0:
            soma += c
    if soma == x:
        print(f'{x} eh perfeito')
    else:
        print(f'{x} nao eh perfeito')
    n -= 1
